### ISSUE TITLE
Shutdown gracefully when reloading configurations

### ISSUE THREAD

**Issue by ben-manes** [jcustenborder/kafka-connect-cdc-mssql#22]
When modifying a connector, Kafka reloads all of them. This one will cause that to fail and halt the service, due to failing to re-register the metrics. As described by [KAFKA-6252](https://issues.apache.org/jira/browse/KAFKA-6252), this occurs when a connector fails to implement `stop()` appropriately.

I suspect the problem is due to the rate limiter. I don't know why it isn't resolved by `stop()` calling `serviceManager.awaitStopped(...)`, or seeming to wait the full minute before reloading. If is it the rate limiter, then increasing the `setRate` in `triggerShutdown` should allow the thread to wake up and terminate quickly.

**Comment by ben-manes** [comment]
Actually, since the thread is asleep it won't wake up immediately. Instead making the stopping logic wait to epochs fixes it.

```java
@Override
protected void triggerShutdown() {
  rateLimiter.acquire();
  rateLimiter.acquire();
}
```

**Comment by jcustenborder** [comment]
@ben-manes [KAFKA-6252](https://issues.apache.org/jira/browse/KAFKA-6252) has been resolved in 1.0.0. Upgrade to this version if you are running 0.10.2.2 or greater for your brokers.
